lms/lmsweb/views.py

```diff
@@ -98,27 +99,30 @@ def ratelimit_handler(e):
     deduct_when=lambda response: response.status_code != 200,
 )
 def login(login_message: Optional[str] = None):
+    next_page = request.form.get('next')
     if current_user.is_authenticated:
-        return get_next_url(request.args.get('next'))
+        return get_next_url(next_page)
 
-    username = request.form.get('username')
-    password = request.form.get('password')
-    next_page = request.form.get('next')
+    form = LoginForm()
     login_message = request.args.get('login_message')
+    if not form.validate_on_submit():
+        return render_template(
+            'login.html', form=form, login_message=login_message,
```

Comment:
Should we keep the login message from the last page request or generate new one? (I might have missed something in the logic here)

Reply:
I don't think I changed anything meaningful here. Me missing something in the logic here would be more probable :P 
From just messing around with the system, it seems to work - what should I fix here?

Reply:
Try to create a form validator instead of the `auth` method and because of that you would remove `113-122` lines.
You can take a look in the `lmsweb/tools/validators.py` and the `auth` method in order to create the validator